crates/rpc/src/handlers/pool.rs

```diff
@@ -55,3 +55,54 @@ pub async fn post_voluntary_exits(
 
     Ok(HttpResponse::Ok())
 }
+
+/// GET /eth/v1/beacon/pool/proposer_slashings
+#[get("/beacon/pool/proposer_slashings")]
+pub async fn get_pool_proposer_slashings(
+    slashing_pool: Data<Arc<SlashingPool>>,
+) -> Result<impl Responder, ApiError> {
+    let slashings = slashing_pool.get_all_proposer_slashings();
+    Ok(HttpResponse::Ok().json(DataResponse::new(slashings)))
+}
+
+/// POST /eth/v1/beacon/pool/proposer_slashings
+#[post("/beacon/pool/post_proposer_slashings")]
+pub async fn post_pool_proposer_slashings(
+    db: Data<ReamDB>,
+    slashing_pool: Data<SlashingPool>,
+    slashing: Json<ProposerSlashing>,
+) -> Result<impl Responder, ApiError> {
+    let highest_slot = db
+        .slot_index_provider()
+        .get_highest_slot()
+        .map_err(|err| {
+            error!("Failed to get_highest_slot, error: {err:?}");
+            ApiError::InternalError
+        })?
+        .ok_or(ApiError::NotFound(
+            "Failed to find highest slot".to_string(),
+        ))?;
+    let mut beacon_state = get_state_from_id(ID::Slot(highest_slot), &db).await?;
+    let proposer_index = slashing.signed_header_1.message.proposer_index;
+
+    if slashing_pool.has_slashing_for_proposer(proposer_index) {
+        return Err(ApiError::BadRequest(
+            "Proposer slashing already exists for this validator".to_string(),
+        ));
+    }
+    let slashing = slashing.into_inner();
+
+    beacon_state
+        .process_proposer_slashing(&slashing)
+        .map_err(|err| {
+            ApiError::BadRequest(format!(
+                "Invalid proposer slashing, it will never pass validation so it's rejected: {err:?}"
+            ))
+        })?;
+    slashing_pool
+        .get_ref()
+        .insert_proposer_slashing(slashing.clone())
+        .map_err(|_| ApiError::BadRequest("Proposer slashing already in pool".to_string()))?;
```

Comment:
```suggestion
        .map_err(|err| ApiError::BadRequest(format!("Proposer slashing already in pool: {err:?}"))?;
```

we shouldn't hide the error, please pass it up